ocaml/xapi/helpers.ml

```diff
@@ -860,26 +860,24 @@ let pool_has_different_host_platform_versions ~__context =
 
 (* Read pool secret if it exists; otherwise, create a new one. *)
 let get_pool_secret () =
-  try
-    Unix.access !Xapi_globs.pool_secret_path [Unix.F_OK] ;
-    pool_secret :=
-      Unixext.string_of_file !Xapi_globs.pool_secret_path
-      |> SecretString.of_string ;
-    Db_globs.pool_secret :=
-      !pool_secret |> SecretString.rpc_of_t |> Db_secret_string.t_of_rpc
-  with _ ->
-    (* No pool secret exists. *)
-    let mk_rand_string () = Uuid.to_string (Uuid.make_uuid ()) in
-    pool_secret :=
-      SecretString.of_string
-      @@ mk_rand_string ()
-      ^ "/"
-      ^ mk_rand_string ()
-      ^ "/"
-      ^ mk_rand_string () ;
-    Db_globs.pool_secret :=
-      !pool_secret |> SecretString.rpc_of_t |> Db_secret_string.t_of_rpc ;
-    SecretString.write_to_file !Xapi_globs.pool_secret_path !pool_secret
+  ( try
+      Unix.access !Xapi_globs.pool_secret_path [Unix.F_OK] ;
+      let ps =
+        Unixext.string_of_file !Xapi_globs.pool_secret_path
+        |> SecretString.of_string
+      in
+      pool_secrets := [ps] ;
+      Db_globs.pool_secret :=
+        ps |> SecretString.rpc_of_t |> Db_secret_string.t_of_rpc
+    with _ ->
+      (* No pool secret exists. *)
+      let module Ptoken = Genptokenlib.Lib in
```

Comment:
`pool_secret` used to be updated in this branch, `pool_secrets` is not. It's not clear to me why this is the case